eng/scripts/New-ReleaseArtifact.ps1

```diff
@@ -0,0 +1,85 @@
+<#
+.SYNOPSIS
+Creates a .tar.gz archive of files in a given folder.
+
+.DESCRIPTION
+This is supposed to help zip up any location where artifacts are properly staged
+for placement in an archive. It does this by reading in the package information
+from the package.json file, then creating an archive named for the package and
+version, and copying the package to an artifact location.
+
+.PARAMETER SourceDirectory
+Source directory to compress. Files and folders in that source directory
+(except .git) will be compressed.
+
+.PARAMETER PackageSpecPath
+Location of the package.json file.
+
+.PARAMETER Destination
+Location to copy the artifact file to.
+
+.PARAMETER Destination
+Name of the destination to which the archive should be copied.
+
+.PARAMETER Workspace
+Workspace folder where assets are staged before creating.
+
+#>
+
+
+param (
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $SourceDirectory,
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $PackageSpecPath,
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $Destination,
+
+    [Parameter()]
+    [ValidateNotNullOrEmpty()]
+    [string] $Workspace = "$env:TEMP/vcpkg-workspace"
+)
+
+$initialLocation = Get-Location
+
+try {
+    $packageSpec = Get-Content -Raw -Path $PackageSpecPath | ConvertFrom-Json
+    $archiveName = $packageSpec.packageName
+
+    Write-Verbose "Archive Name: $archiveName"
+
+    # Set up workspace, copy source to workspace
+    Remove-Item -Path $Workspace -Force -Recurse -ErrorAction Ignore
```

Comment:
Is this the part that the C++ team will need to change if they want to scope changes to just one library as opposed to the entire repo? Or do you expect that to be controlled by the incoming SourceDirectory parameter?

Reply:
Workspace should not be updated. I set it directly in the script.

Reply:
I wasn't referring to workspace but SourceDirectory as I was trying to figure out what the team is supposed to update once they decide to stop shipping the entire repo for each package.